tools/crm_mon.c

```diff
@@ -177,6 +164,267 @@ crm_trigger_t *refresh_trigger = NULL;
+      "Group resources by node",
+      NULL },
+
+    { "inactive", 'r', 0, G_OPTION_ARG_NONE, &options.inactive_resources,
+      "Display inactive resources",
+      NULL },
+
+    { "failcounts", 'f', G_OPTION_FLAG_NO_ARG, G_OPTION_ARG_CALLBACK, show_failcounts_cb,
+      "Display resource fail counts",
+      NULL },
+
+    { "operations", 'o', G_OPTION_FLAG_NO_ARG, G_OPTION_ARG_CALLBACK, show_operations_cb,
+      "Display resource operation history",
+      NULL },
+
+    { "timing-details", 't', G_OPTION_FLAG_NO_ARG, G_OPTION_ARG_CALLBACK, show_operations_cb,
+      "Display resource operation history with timing details",
+      NULL },
+
+    { "tickets", 'c', G_OPTION_FLAG_NO_ARG, G_OPTION_ARG_CALLBACK, show_tickets_cb,
+      "Display cluster tickets",
+      NULL },
+
+    { "watch-fencing", 'W', 0, G_OPTION_ARG_NONE, &options.watch_fencing,
+      "Listen for fencing events. For use with --external-agent",
+      NULL },
+
+    { "fence-history", 'm', 0, G_OPTION_ARG_INT, &options.fence_history_level,
+      "Show fence history:\n"
+      INDENT "0=off, 1=failures and pending (default without option),\n"
+      INDENT "2=add successes (default without value for option),\n"
+      INDENT "3=show full history without reduction to most recent of each flavor",
+      "LEVEL" },
+
+    { "neg-locations", 'L', 0, G_OPTION_ARG_CALLBACK, show_bans_cb,
+      "Display negative location constraints [optionally filtered by id prefix]",
+      NULL },
+
+    { "show-node-attributes", 'A', G_OPTION_FLAG_NO_ARG, G_OPTION_ARG_CALLBACK, show_attributes_cb,
+      "Display node attributes",
+      NULL },
+
+    { "hide-headers", 'D', G_OPTION_FLAG_NO_ARG, G_OPTION_ARG_CALLBACK, hide_headers_cb,
+      "Hide all headers",
+      NULL },
+
+    { "show-detail", 'R', 0, G_OPTION_ARG_NONE, &options.print_clone_detail,
+      "Show more details (node IDs, individual clone instances)",
+      NULL },
+
+    { "brief", 'b', 0, G_OPTION_ARG_NONE, &options.print_brief,
+      "Brief output",
+      NULL },
+
+    { "pending", 'j', G_OPTION_FLAG_HIDDEN, G_OPTION_ARG_NONE, &options.print_pending,
+      "Display pending state if 'record-pending' is enabled",
+      NULL },
+
+    { NULL }
+};
+
+static GOptionEntry mode_entries[] = {
+    { "as-html", 'h', G_OPTION_FLAG_FILENAME, G_OPTION_ARG_CALLBACK, as_html_cb,
+      "Write cluster status to the named HTML file",
+      "FILE" },
+
+    { "as-xml", 'X', G_OPTION_FLAG_NO_ARG, G_OPTION_ARG_CALLBACK, as_xml_cb,
+      "Write cluster status as XML to stdout. This will enable one-shot mode.",
+      NULL },
+
+    { "web-cgi", 'w', G_OPTION_FLAG_NO_ARG, G_OPTION_ARG_CALLBACK, as_cgi_cb,
+      "Web mode with output suitable for CGI (preselected when run as *.cgi)",
+      NULL },
+
+    { "simple-status", 's', G_OPTION_FLAG_NO_ARG, G_OPTION_ARG_CALLBACK, as_simple_cb,
+      "Display the cluster status once as a simple one line output (suitable for nagios)",
+      NULL },
```

Comment:
FYI, simple-status won't have its own formatter (it can use text). It's not so much a format as alternative content. Long-term, I want to separate that into its own tool and deprecate it here.